src/connection_wrap.h

```diff
@@ -0,0 +1,22 @@
+#ifndef SRC_CONNECTION_WRAP_H_
+#define SRC_CONNECTION_WRAP_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "env.h"
+#include "v8.h"
+
+namespace node {
+
+class ConnectionWrap {
+ protected:
+  template<typename WrapType, typename UVType>
```

Comment:
It might be nicer to have the `template`ing done on the class itself, `class A : public B<A> { … };` isn’t that uncommon of a pattern + it would allow lifting the `handle_` to `ConnectionWrap` as `UVType handle_;` (I think?).